hathor/merged_mining/coordinator.py

```diff
@@ -719,12 +719,11 @@ async def submit_to_bitcoin(self, job: SingleMinerJob, work: SingleMinerWork) ->
                     'work': work,
                 }, f)
                 f.close()
-                self.log.warn('Work rejected from Bitcoin Fullnode. Relevant job+work dumped.',
-                              code=res, filename=f.name)
+                self.log.warn('block rejected from Bitcoin', code=res, dump_file=f.name)
             else:
-                self.log.warn('Work rejected from Bitcoin Fullnode.', code=res)
+                self.log.warn('block rejected from Bitcoin', code=res)
         else:
-            self.log.info('Bitcoin RPC submit response', res=res)
+            self.log.info('other result from Bitcoin', res=res)
```

Comment:
Maybe "unexpected result from Bitcoin"?

Reply:
I'm not sure if it's unexpected. If it's not `None` (success) or `'high-hash'` they could all be called `unexpected` because it would signal an implementation error, actually even `'high-hash'` would be unexpected because of the `if block_hash.to_u256() > block_target.to_u256()` condition above. Maybe I could refactor these since I understand these errors better now, and because we'll validate the template before sending the work to the miner (which is on another branch).

Reply:
That would be great! You could also include a list of the known responses from the Bitcoin full node.

Reply:
OK, I'll do this on `feat/merged-mining-better-api` where I'm already touching this part. Also, although it might be helpful to list known responses and symptoms, it's probably not that useful. It should always return `None`, otherwise there is an implementation error. At least that's what I get from all the errors I've encountered. But I'll add a brief list anyway so I can remember it better in case there is a regression.

Reply:
Here: https://github.com/HathorNetwork/hathor-core/pull/114/files#diff-eb36e58f1ad6ea122750ed239d9cd974L709-L729